esphome/components/ethernet/ethernet_component.cpp

```diff
@@ -356,6 +358,21 @@ std::string EthernetComponent::get_use_address() const {
 
 void EthernetComponent::set_use_address(const std::string &use_address) { this->use_address_ = use_address; }
 
+bool EthernetComponent::powerdown() {
```

Comment:
This should just be in `on_shutdown` instead of making another function for it.
This should never be called during normal runtime anyway as it will mess things up.
```suggestion
void EthernetComponent::on_shutdown() {
```

Reply:
I did it this way because there might eventually be different / additional things to do on shutdown events before or after the powerdown method (which really should only send the command to the PHY controller), like do something about the network stack. I my use case there is a need to issue `powerdown()` in some situations to save a bit of power without going into full deep sleep; the user code can reboot the processor, go into deep sleep, or call setup() again if needed. I really do not see any harm in having a separate event handler `on_shutdown()`  and a potentially user-callable / inheritable method `powerdown()`. People who go messing around the C++ API should be able to assess the consequences of calling `powerdown()`.

Reply:
There also might be a need to check whether the PHY really went into powerdown or some error happened, in which case we could try again, alert the user or whatever.

Reply:
Ok, this can stay as is then